narwhals/_arrow/utils.py

```diff
@@ -452,3 +455,40 @@ def _parse_time_format(arr: pa.Array) -> str:
         if pc.all(matches.is_valid()).as_py():
             return time_fmt
     return ""
+
+
+def _window_agg(
```

Comment:
we're still iterating in Python here...dunno, maybe we should just raise NotImplementedError here?

curious to see a timing comparison of:
- convert to pandas, do rolling mean, back to pyarrow
- do it with this function

Reply:
I will give it a shot and report back :)

Reply:
Yep this is definitly orders of magnitude slower. I will try with a specified implementation